common/generate_features.py

```diff
@@ -65,9 +65,15 @@ def __init__(self,
         self.num_emails = len(self.emails)
 
         #Convert from percentages to number of emails in each
-        self.sender_profile_num_emails = int(self.sender_profile_percentage * self.num_emails)
-        self.data_matrix_num_emails = int(self.data_matrix_percentage * self.num_emails)
-        self.test_matrix_num_emails = int(self.test_matrix_percentage * self.num_emails)
+        if self.num_emails <= 1:
```

Comment:
For the case when self.num_emails == 1, it looks like the code will not assign that single email to any of the 3 categories. I was under the impression that in this case, we would send it to the test_matrix.